test/module/irohad/ametsuchi/postgres_query_executor_test.cpp

```diff
@@ -1398,6 +1493,142 @@ namespace iroha {
+      auto size = 2;
+      auto query_response =
+          this->queryPage(size, std::nullopt, std::nullopt, std::nullopt, 1);
+      checkSuccessfulResult<TransactionsPageResponse>(
+          std::move(query_response),
+          [this, size](const auto &tx_page_response) {
+            EXPECT_EQ(tx_page_response.transactions().size(), size);
+          });
+    }
+
+    /**
+     * @given initialized storage, user has 10 transactions committed
+     * @when query contains 15 page size
+     * @and first block time is before transactions
+     * @and last block time is after transactions
+     * @and first block height is 2
+     * @and last block height is 5
+     * @then response contains 2 committed transactions
+     */
+    TYPED_TEST(GetPagedTransactionsExecutorTest,
+               FirstTimeLastTimeFirstHeightLastHeightSpecified) {
+      auto first_tx_time = iroha::time::now();
+      this->createTransactionsAndCommit(10, true);
+      auto last_tx_time = iroha::time::now();
+      auto size = 2;
+      auto query_response = this->queryPage(
+          size, std::nullopt, first_tx_time, last_tx_time, 2, 5);
+      checkSuccessfulResult<TransactionsPageResponse>(
+          std::move(query_response),
+          [this, size](const auto &tx_page_response) {
+            EXPECT_EQ(tx_page_response.transactions().size(), size);
+          });
+    }
+
+    /**
+     * @given initialized storage, user has 10 transactions committed
+     * @when query contains 15 page size
+     * @and last block height is 5
+     * @then response contains 4 committed transactions
+     */
+    TYPED_TEST(GetPagedTransactionsExecutorTest, LastHeightSpecified) {
+      this->createTransactionsAndCommit(10, true);
+      auto size = 4;
+      auto query_response = this->queryPage(
+          size, std::nullopt, std::nullopt, std::nullopt, std::nullopt, 5);
+      checkSuccessfulResult<TransactionsPageResponse>(
+          std::move(query_response),
+          [this, size](const auto &tx_page_response) {
+            EXPECT_EQ(tx_page_response.transactions().size(), size);
+          });
+    }
+
+    /**
+     * @given initialized storage, user has 10 transactions committed
+     * @when query contains 15 page size
+     * @and first block height is 2
+     * @and last block height is 5
+     * @then response contains 2 committed transactions
+     */
+    TYPED_TEST(GetPagedTransactionsExecutorTest, FirstAndLastHeightSpecified) {
+      this->createTransactionsAndCommit(10, true);
+      auto size = 2;
+      auto query_response =
+          this->queryPage(size, std::nullopt, std::nullopt, std::nullopt, 2, 5);
+      checkSuccessfulResult<TransactionsPageResponse>(
+          std::move(query_response),
+          [this, size](const auto &tx_page_response) {
+            EXPECT_EQ(tx_page_response.transactions().size(), size);
+          });
+    }
     // --------------------\ end of tx pagination tests /-------------------- //
```

Comment:
Some tests are missing - You are testing 10 transactions and filtering transactions before all of them and after all of them with first_tx_time and last_tx_time - it is working. But it is not testing if Your code is really working. So I suggest that test:
1. Commit 3 transactions
2. Set first_tx_time
3. Commit 4 transactions
4. set last_tx_time
5. Commit 5 transactions
6. As a result You should have 4 transactions.

Reply:
I  solved it by passing first and last time by reference, and than using it in query. I also have to sleep the thread for few miliseconds to get different timestamps of transactions

Reply:
OK